### DISCUSSION TITLE
LATENZA Sonoff movimento

### DISCUSSION THREAD

**Discussion by nromagna** [indomus/forum#424] [HUB: Home Assistant]
Ciao a tutti!
Da poco mi sono avvicinato ad HA e ne sono rimasto piacevolmente colpito. Ho integrato un po' tutto a casa ed ora mi sto divertendo un sacco con gli automatismi! 
Ne ho creato uno **MOLTO banale**, ossia se sensore movimento Sonoff passa da off ad on si accende la luce per x sec. e fin qua tutto bene... però se resto nel raggio d'azoine del sensore o, se peggio ancora, me ne vado e la luce, giustamente si spegne,  poi ritorno NON si accende la luce (presumo, colpa della latenza).
Come posso risolvere?
Grazie mille a tutti ^_^

```- id: 'xxxxxxx'
  alias: Movimento credenza
  description: ''
  trigger:
  - platform: state
    from: 'off'
    to: 'on'
    entity_id: binary_sensor.sonoff_a48001xxx
  condition: []
  action:
  - scene: scene.credenza_on
  - wait_template: ''
    timeout: 00:00:25
  - scene: scene.credenza_off
  mode: single

**Comment by KC-inDomus** [comment]
usa delay anziché wait_template, e imposta  mode: restart per questa automazione

**Comment by nromagna** [comment]
Grazie mille @KC-inDomus questa sera provo. :)

**Comment by nromagna** [comment]
Nulla da fare :_(
sente il movimento si accende, esco dalla zona d'azione passa tempoi si spegne ma se rientro nella zona d'aziopne NON si riaccende...
[new 2.txt](https://github.com/indomus/forum/files/6653385/new.2.txt)

**Reply by snakuzzo** [reply]
per favore non scrivere sempre thread diversi. Fai reply sotto l'ultima risposta. Grazie.
L'unica cosa che mi viene in mente è che lo stato del sonoff (e non del sensore su HA) sia ancora `on` quando ripassi e quindi non scatta il trigger su HA.
Non ho capito bene che tipo di sensore stai usando però.
E' Zigbee o WIFI ? E' sempre connesso oppure va in deep sleep ?

**Reply by nromagna** [reply]
Certo scusami.
ZB (SNZB-03) collegato ad un Sonoff Mini sempre ZB.

La seconda prova è stata quella di rimanere sotto il sensore (e quindi in teoria dovrebbe rimanere su ON) ma la luce si è spenta ugualmente dopo i 20sec, Quindi sono uscito dalla stanza, rientrato ma nulla, non si accende.
Si riaccende solo se rientro nella stanza dopo x minuti...

PS
La stessa cosa mi capita con un altro sensore di movimento (stesso modello) ma collegato ad una strip led Hue, davo la colpa alla differenza di marche ma a quanto pare non è così...

**Reply by Scialla** [reply]
dipende dalla latenza del sensore, ci va qualche minuto prima che torni in off per poi tornare a on col movimento

**Reply by nromagna** [reply]
Si può diminuire la latenza oppure potrebbe causare problemi?

Un sensore di movimento Hue che ho nella zone notte collegato con una lampadina sempre Hue non ha di questi problemi, certo costa anche 4 volte di più... 

E poi perchè un sensore "porta aperta" sempre Sonoff funziona senza latenze?

Scusate le mille domande ma è uno dei pochi posti dove sono riuscito a trovare il 90% delle risposte :)

**Reply by Scialla** [reply]
Sono caratteristiche dei sensori PIR, quelli porta non hanno latenze, in un certo senso sono "volute". Quelli Aqara sono modificabili via HW con una matita, i Sonoff non saprei ma non credo.

**Reply by snakuzzo** [reply]
con quelli Aqara di default è 60s ... modificandolo si porta a 5s.
con i sonoff non saprei proprio

Ad ogni modo, se usi il ZB Bridge con firmware originale, controlla cosa accade sull'app

**Reply by nromagna** [reply]
Non ci siamo... capisco la latenza ma ci dovrebbe essere qualcosa di erraro nell'automatismo. Ieri ero nella stanza proprio accanto al sensore, sono passati i 20 secondi e si è spento. Non dovrebbe ripartire il conto alla rovescia nel caso si ulteriori movimenti?
Grazie.
